nemo/collections/nlp/modules/common/megatron/attention.py

```diff
@@ -53,6 +58,19 @@
 
     HAVE_MEGATRON_CORE = False
 
+try:
+    from flash_attn.bert_padding import pad_input, unpad_input
+    from flash_attn.flash_attn_interface import flash_attn_unpadded_func
+    from flash_attn.flash_attn_triton import flash_attn_func
+
+except (ImportError, ModuleNotFoundError):
+    logging.warning(
+        "flash_attn was not found. Please see the installation instructions: https://github.com/HazyResearch/flash-attention."
+        "If you use flash_attn with triton. Please install triton==2.0.0.dev20221202."
+    )
+    flash_attn_unpadded_func, flash_attn_func = None, None
+    unpad_input, pad_input = None, None
```

Comment:
You might want to add a `HAVE_FLASH_ATTENTION` variable here, then raise an error if a class that needs it is instantiated and you don't have it

Reply:
We can also do a config check very on in megatron_base_model.py if user configure `use_flash_attention=True` but `HAVE_FLASH_ATTENTION=False` then raise an error

Reply:
Added